prov/sm2/src/sm2.h

```diff
@@ -76,6 +76,8 @@
 #define SM2_PREFIX_NS		"fi_ns://"
 #define SM2_VERSION		1
 #define SM2_IOV_LIMIT		4
+#define SM2_INJECT_USER_DATA_SIZE \
+	(SM2_INJECT_SIZE - sizeof(struct sm2_xfer_header))
```

Comment:
I think this makes the difference between SM2_INJECT_SIZE and SM2_INJECT_USER_DATA_SIZE confusing. I would leave this one as SM2_INJECT_SIZE (which I read as the size of message which you can inject) and the other one to something like SM2_XFER_ENTRY_SIZE